app/controllers/challenge_partnerships_controller.rb

```diff
@@ -25,39 +27,34 @@ def success; end
 private
 
   def authorize_partnership
-    @token = params[:token] || params.dig(:challenge_partnership_form, :token)
-    partnership_id = params[:partnership] || params.dig(:challenge_partnership_form, :partnership_id)
-    if @token.present?
-      notification_email = PartnershipNotificationEmail.find_by(token: @token)
-      raise ActionController::RoutingError, I18n.t(:not_found) if notification_email.blank?
-
-      @partnership = notification_email.partnership
-    elsif partnership_id.present?
-      @partnership = Partnership.find(partnership_id)
-      authorize(@partnership, :update?)
+    if access_token.present?
+      if access_token.school != partnership.school || !access_token.permits?(:challenge_partnership)
+        raise Pundit::NotAuthorizedError
+      end
     else
-      raise ActionController::RoutingError, I18n.t(:not_found)
+      authorize(partnership, :update?)
```

Comment:
I'm not sure what's going on here, what is `current_user` going to be?

Reply:
`current_user` returns a user who is signed in. When accessing any page with a token, `current_user` is nil, the same as with NominationEmail tokens. I've considered creating `AnonymousSchoolUser` object to use as a `current_user` for token access, however it would require quite a strange interface to mimic some of the association required in policies.

But maybe it would not be that hard, happy to revisit?